eav/queryset.py

```diff
@@ -282,3 +282,78 @@ def get(self, *args, **kwargs):
         the ``Manager`` get method.
         """
         return super(EavQuerySet, self).get(*args, **kwargs)
+
+    def order_by(self, *fields):
+        # Django only allows to order querysets by direct fields and
+        # foreign-key chains. In order to bypass this behaviour and order
+        # by EAV attributes, it is required to construct custom order-by
+        # clause manually using Django's conditional expressions.
+        # This will be slow, of course.
+        order_clauses = []
+        query_clause = self
+
+        for term in [t.split('__') for t in fields]:
+            # Continue only for EAV attributes.
+            if len(term) == 2 and term[0] == 'eav':
+                # Retrieve Attribute over which the ordering is performed.
+                attr = Attribute.objects.get(slug=term[1])
```

Comment:
Raises DoesNotExist and not AttributeError as you mentioned in your other comment. Is that correct?

Reply:
I would leave `DoesNotExist` but change the exception message to something more informative, such as "Specified attribute value doesn't exist".

Reply:
Is this a common now that there are `DoesNotExist` classes linked somewhere outside a particular model?